data-prepper-api/src/main/java/com/amazon/dataprepper/model/processor/AbstractProcessor.java

```diff
@@ -41,6 +41,13 @@ public AbstractProcessor(final PluginSetting pluginSetting) {
         timeElapsedTimer = pluginMetrics.timer(MetricNames.TIME_ELAPSED);
     }
 
+    protected AbstractProcessor(final PluginMetrics pluginMetrics) {
+        this.pluginMetrics = pluginMetrics;
+        recordsInCounter = pluginMetrics.counter(MetricNames.RECORDS_IN);
```

Comment:
nit: move line 46-48 and line 39-41 to private constructor and invoke private constructor.